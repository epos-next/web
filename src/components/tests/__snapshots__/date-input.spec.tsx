import DateInput, { formatSelectedDate } from "@components/date-input";
import DateHelper from "@helpers/date-helper";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import "jest-styled-components"
import '@testing-library/jest-dom'


it("should math snapshot", () => {
    const { asFragment } = render(<DateInput/>)
    expect(asFragment()).toMatchSnapshot()
});

it("should show calendar if user focus input field", () => {
    const { getByTestId } = render(<DateInput/>)
    fireEvent.focus(getByTestId("input-base"))
    expect(getByTestId("calendar-container")).toHaveAttribute("data-show", "true")
});

it("should call callback if user click on date field", () => {
    const callback = jest.fn()
    const now = new Date(2021, 11, 21)
    jest.spyOn(DateHelper, "now", "get").mockReturnValue(now)

    const { getByTestId } = render(<DateInput onChange={ callback }/>)
    fireEvent.focus(getByTestId("input-base"))

    const date = new Date(2021, 10, 30)
    fireEvent.click(getByTestId(`date-${ date.toISOString() }`))

    expect(callback).toHaveBeenCalledWith(date)

    jest.clearAllMocks()
});

it("should call close calendar if user click on date field", () => {
    const callback = jest.fn()
    const now = new Date(2021, 11, 21)
    jest.spyOn(DateHelper, "now", "get").mockReturnValue(now)

    const { getByTestId } = render(<DateInput onChange={ callback }/>)
    fireEvent.focus(getByTestId("input-base"))

    const date = new Date(2021, 10, 30)
    fireEvent.click(getByTestId(`date-${ date.toISOString() }`))

    expect(getByTestId("calendar-container")).toHaveAttribute("data-show", "false")

    jest.clearAllMocks()
});

it("should use state date value to display in input if no props value", () => {
    const callback = jest.fn()
    const now = new Date(2021, 11, 21)
    jest.spyOn(DateHelper, "now", "get").mockReturnValue(now)

    const { getByTestId } = render(<DateInput onChange={ callback }/>)

    expect(getByTestId("input-base")).toHaveValue("21.12.2021")

    fireEvent.focus(getByTestId("input-base"))

    const date = new Date(2021, 10, 30)
    fireEvent.click(getByTestId(`date-${ date.toISOString() }`))

    expect(getByTestId("input-base")).toHaveValue("30.11.2021")

    jest.clearAllMocks()
});

it("should use props value", () => {
    const propValue = new Date(2021, 8, 1);
    const now = new Date(2021, 11, 21)
    jest.spyOn(DateHelper, "now", "get").mockReturnValue(now)

    const { getByTestId } = render(<DateInput value={propValue}/>)

    expect(getByTestId("input-base")).toHaveValue("01.09.2021")

    fireEvent.focus(getByTestId("input-base"))

    const date = new Date(2021, 10, 30)
    fireEvent.click(getByTestId(`date-${ date.toISOString() }`))

    expect(getByTestId("input-base")).toHaveValue("01.09.2021")

    jest.clearAllMocks()
});

describe("testing formatSelectedDate function", () => {
    it("should format simple date", () => {
        const date = new Date(2021, 10, 15)
        expect(formatSelectedDate(date)).toEqual("15.11.2021")
    });

    it("should add \"0\" before one symbol month and date", () => {
        const date = new Date(2021, 1, 4)
        expect(formatSelectedDate(date)).toEqual("04.02.2021")
    });

    it("should return empty string if input is null", () => {
        expect(formatSelectedDate(null)).toEqual("")
    });
})




