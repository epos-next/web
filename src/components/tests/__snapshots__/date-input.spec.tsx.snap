// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should math snapshot 1`] = `
<DocumentFragment>
  .c13 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  color: white;
}

.c13:after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  margin: 0 auto;
  border-radius: 50%;
  background: linear-gradient(328.45deg,#656DFD 11.68%,#949EFD 85.36%);
  width: 36px;
  height: 36px;
  z-index: -1;
}

.c12 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--primary);
  cursor: pointer;
}

.c11 {
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--light-primary);
  cursor: pointer;
}

.c9 {
  display: grid;
  grid-template-columns: repeat(7,1fr);
  margin-top: 5px;
}

.c10 {
  font-size: 15px;
  line-height: 18px;
  color: var(--secondary);
  width: 100%;
  height: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 15px;
  line-height: 18px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  font-weight: 500;
  font-size: 18px;
  line-height: 22px;
  color: var(--primary);
}

.c8 {
  font-size: 18px;
  line-height: 22px;
  color: var(--primary);
  -webkit-appearance: none;
  -moz-appearance: none;
  text-indent: 1px;
  text-overflow: '';
  outline: none;
  text-align: right;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 calc((100% / 7 - 18px) / 2);
}

.c5 {
  width: 282px;
}

.c2 {
  width: 100%;
  border: 2px solid var(--light-primary);
  border-radius: 10px;
  outline: none;
  padding: 14px;
  font-size: 18px;
  color: var(--primary);
  -webkit-transition: 150ms border;
  transition: 150ms border;
}

.c2::-webkit-input-placeholder {
  color: var(--light-primary);
}

.c2::-moz-placeholder {
  color: var(--light-primary);
}

.c2:-ms-input-placeholder {
  color: var(--light-primary);
}

.c2::placeholder {
  color: var(--light-primary);
}

.c2:focus {
  border: 2px solid var(--contrast);
}

.c2[data-error=true] {
  color: var(--error);
  border: 2px solid var(--error);
}

.c2[data-error=true]:focus {
  border: 2px solid var(--error);
}

.c1 {
  width: 100%;
  position: relative;
}

.c4 {
  padding: 10px;
  border-radius: 10px;
  background-color: white;
  border: 1px solid var(--light-primary);
  position: absolute;
  top: calc(100% + 10px);
  left: 0;
  z-index: 100000;
  -webkit-transition: -webkit-transform 300ms ease,opacity 300ms ease;
  -webkit-transition: transform 300ms ease,opacity 300ms ease;
  transition: transform 300ms ease,opacity 300ms ease;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
}

.c4[data-show=false] {
  opacity: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transition-delay: 400ms;
  transition-delay: 400ms;
}

.c3 {
  position: absolute;
  top: 14px;
  right: 14px;
}

.c0 {
  width: 100%;
  position: relative;
}

@media screen and (max-width:960px) {
  .c5 {
    width: calc(100%);
  }
}

@media screen and (max-width:768px) {
  .c2 {
    padding: 13px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 input "
      data-testid="container"
    >
      <input
        class="c2"
        data-error="false"
        data-testid="input-base"
        readonly=""
        type="text"
        value="13.12.2021"
      />
    </div>
    <img
      class="c3"
      src="/icons/calendar.png"
    />
    <div
      class="c4"
      data-show="false"
      data-testid="calendar-container"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <h5
            class="c7"
          >
            Календарь
          </h5>
          <select
            class="c8"
            data-testid="month"
            dir="rtl"
          >
            <option
              value="0"
            >
              Сентябрь
            </option>
            <option
              value="1"
            >
              Октябрь
            </option>
            <option
              value="2"
            >
              Ноябрь
            </option>
            <option
              value="3"
            >
              Декабрь
            </option>
            <option
              value="4"
            >
              Январь
            </option>
            <option
              value="5"
            >
              Февраль
            </option>
            <option
              value="6"
            >
              Март
            </option>
            <option
              value="7"
            >
              Апрель
            </option>
            <option
              value="8"
            >
              Май
            </option>
            <option
              value="9"
            >
              Июнь
            </option>
            <option
              value="10"
            >
              Июль
            </option>
            <option
              value="11"
            >
              Август
            </option>
          </select>
        </div>
        <div
          class="c9"
        >
          <div
            class="c10"
          >
            Пн
          </div>
          <div
            class="c10"
          >
            Вт
          </div>
          <div
            class="c10"
          >
            Ср
          </div>
          <div
            class="c10"
          >
            Чт
          </div>
          <div
            class="c10"
          >
            Пт
          </div>
          <div
            class="c10"
          >
            Сб
          </div>
          <div
            class="c10"
          >
            Вс
          </div>
          <div
            class="c11"
            data-testid="date-2021-11-28T19:00:00.000Z"
          >
            29
          </div>
          <div
            class="c11"
            data-testid="date-2021-11-29T19:00:00.000Z"
          >
            30
          </div>
          <div
            class="c12"
            data-testid="date-2021-11-30T19:00:00.000Z"
          >
            1
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-01T19:00:00.000Z"
          >
            2
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-02T19:00:00.000Z"
          >
            3
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-03T19:00:00.000Z"
          >
            4
          </div>
          <div
            class="c11"
            data-testid="date-2021-12-04T19:00:00.000Z"
          >
            5
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-05T19:00:00.000Z"
          >
            6
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-06T19:00:00.000Z"
          >
            7
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-07T19:00:00.000Z"
          >
            8
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-08T19:00:00.000Z"
          >
            9
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-09T19:00:00.000Z"
          >
            10
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-10T19:00:00.000Z"
          >
            11
          </div>
          <div
            class="c11"
            data-testid="date-2021-12-11T19:00:00.000Z"
          >
            12
          </div>
          <div
            class="c13"
            data-testid="date-2021-12-12T19:00:00.000Z"
          >
            13
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-13T19:00:00.000Z"
          >
            14
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-14T19:00:00.000Z"
          >
            15
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-15T19:00:00.000Z"
          >
            16
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-16T19:00:00.000Z"
          >
            17
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-17T19:00:00.000Z"
          >
            18
          </div>
          <div
            class="c11"
            data-testid="date-2021-12-18T19:00:00.000Z"
          >
            19
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-19T19:00:00.000Z"
          >
            20
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-20T19:00:00.000Z"
          >
            21
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-21T19:00:00.000Z"
          >
            22
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-22T19:00:00.000Z"
          >
            23
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-23T19:00:00.000Z"
          >
            24
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-24T19:00:00.000Z"
          >
            25
          </div>
          <div
            class="c11"
            data-testid="date-2021-12-25T19:00:00.000Z"
          >
            26
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-26T19:00:00.000Z"
          >
            27
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-27T19:00:00.000Z"
          >
            28
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-28T19:00:00.000Z"
          >
            29
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-29T19:00:00.000Z"
          >
            30
          </div>
          <div
            class="c12"
            data-testid="date-2021-12-30T19:00:00.000Z"
          >
            31
          </div>
          <div
            class="c11"
            data-testid="date-2021-12-31T19:00:00.000Z"
          >
            1
          </div>
          <div
            class="c11"
            data-testid="date-2022-01-01T19:00:00.000Z"
          >
            2
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
